newsweek.com home page clone, merge request

8Bts opened the pull request:
Thanks for checking out my project!

Here is the briefing before reviewing the code:

### BASIC STRUCTURE

3 semantic elements are used in order to build a basic HTML structure:

1.  `<nav>` element for the navigation bar contents.
2.  `<main>` element for the sections between the navigation bar and footer section.
3.  `<footer` element for footer section contents.

### LAYOUT TECHNIQUES

To achieve same layout of the actual page, I used the following layout modules:

- **Floats** for building basic layout of main-contents container, that is between the navbar and in the magazine section, and some inner boxes.

- **Bootstrap Grid System**  for basic containers and inner boxes.

- **CSS Grid** for basic containers and inner boxes.

- **Flex Box** by bootstrap and custom classes, for basic containers and inner boxes.

Please review my project before I merge my branch into master and don't forget to give stars ⭐⭐⭐   👍

8Bts committed:
Added initial files , build empty structure

8Bts committed:
Added top navbar contents, aligned them

8Bts committed:
Added logo image

8Bts committed:
Completed top of navbar: Styled elements color, spacing, size, typography

8Bts committed:
Completed nav bar section

8Bts committed:
Added container for the main section

8Bts committed:
Defined a responsible and accurately proportioned structure for main contents section

8Bts committed:
Added and styled contents of col1-left

8Bts committed:
Added and styled content of middle col

8Bts committed:
Added and styled contents of OPINION and DEBATE sections

8Bts committed:
Added and styled right columns contents, gave order on mobile size

8Bts committed:
Added contents of sponsored 2 section

8Bts committed:
Added js to change the height of more-stories-section and added position sticky

8Bts committed:
Added and styled contents of in the magazine section

8Bts committed:
Added and styled editor's picks contents

8Bts committed:
Added and styled contents of last articles section

8Bts committed:
Added and styled contents of the footer section

8Bts committed:
Added readme details

8Bts committed:
Updated readme.md

MarioDena reviewed (changes requested):
Hello! Great work on your project! 👍  I can tell you pay attention to detail on your work.

However your CSS is not following good practices. 

## Changed required 

- [ ] Please clean up your CSS as there are repeated rules, for example `display: inline-block` is used 15 times.  here is an article for help: https://speckyboy.com/good-bad-css-practices/

## Optional Changes

- [ ]  I recommend replacing the screenshot on your readme file, it's quite long and if someone is interested in for example what technologies you are using your screenshot is on the way of easy navigation.

After making the requested changes please resubmit your project for another review
Cheers!

## Code Review by Mario Enrique Delgado Naranjo

8Bts committed:
Fixed strecthing images of in magazine section, cleared css as much as possible

8Bts committed:
Update README.md

8Bts commented:
Thank you for your review @MarioDena. I checked my CSS and cleared it as much as I could. The reason for using the same properties many times is because of elements grouping. Those properties are applied to the element groups that are faster and more clear than using utility classes and applying them on HTML one by one.

Cheers!

8Bts committed:
Added flex-wrap to footer links

8Bts committed:
Merge branch 'home-page' of https://github.com/8Bts/newsweek-clone into home-page

8Bts committed:
readme update

trekab reviewed (changes requested):
Well done @8Bts, :clap:. You are almost there. I have noticed that there are several scenarios in your CSS where you could have relied on bootstrap classes but didn't. For example when it came to styling margins, borders, padding, use of the display property, use of Flex, and many others. It should be noted that the main point of this project is to remember to use bootstrap classes wherever possible([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png)) for example [Flex](https://getbootstrap.com/docs/4.3/layout/utilities-for-layout/#flexbox-options). 

__Kindly refine your CSS to use more bootstrap classes and submit another code review request__
_Feel free to contact me on slack for any inquiries about this review_
__Treasure Kabareebe__

8Bts committed:
Added more bootsrap classes that replace css properties

rammazzoti2000 commented on index.html:
```diff
@@ -0,0 +1,1433 @@
+              </span>
+            </div>
+            <div class="nav-bottom-right">
+              <span class="search-span">Search</span>
```
I noticed that here you used plain text, this is supposed to be an input please fix this

8Bts replied:
+

8Bts replied:
🆗

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+.nav-top {
+  background-color: #f72210;
+  height: 50px;
+  align-items: center;
```
can be used as bootstrap class
`align-items-center`

8Bts replied:
🆗

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+}
+
+.nav-top-right span {
+  color: white;
+  font-weight: 600;
```
can be used as bootstrap classes
`font-weight-bold`

8Bts replied:
🆗

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+  color: var(--prime-color);
+  font-weight: 600;
+  font-size: 1.15rem;
+  background-color: white;
```
bootstrap class
`font-weight-bold bg-light`

8Bts replied:
🆗

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+  background-color: transparent;
+  color: #fb9a94;
+  font-size: 1.8rem;
+  position: relative;
```
bootstrap class
`position-relative bg-transparent`

8Bts replied:
Unfortunately, there are no such classes in bootstrap

8Bts replied:
🆗

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+.nav-bottom {
+  padding: 1em 0.2em;
+  padding-right: 158px;
+  position: relative;
```
bootstrap class
`position-relative`

8Bts replied:
no such class

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+}
+
+.nav-bottom .nav-bottom-links {
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: center;
+  width: 100%;
```
bootstrap classes:
`d-flex flex-row flex-nowrap justify-content-center w-100`

8Bts replied:
This one is applied to 11 `<span>` elements.

rammazzoti2000 replied:
This is a bootstrap project, no discussion, no compromises.

rammazzoti2000 commented on css/style.css:
```diff
@@ -0,0 +1,928 @@
+/* =========================================== Main Contents Section ====================================== */
+
+.main-contents {
+  position: relative;
+  padding: 0;
+  margin-top: 2px;
```
bootstrap classes:
`position-relative p-0 mt-3`

8Bts replied:
This selection targets 28 elements.

rammazzoti2000 replied:
This is a bootstrap project, no discussion, no compromises.

rammazzoti2000 reviewed (changes requested):
### Hello @8Bts :raised_hands: your project looks very good, the attention to small details and alignments is impressive, good job :clap: 
Also great job on completing the changes required by the previous TSE's.
There are still some small fixes that are preventing me to allow this project
### Required Changes :construction: :construction_worker: 
- [x] Stickler must be enabled :red_circle: 
- [x] Although you added bootstrap classes to your project is not enough, you can do much better, take a look at the inline comments, those are just a few examples I added, but you can use more, please fix that :pray: :point_right: ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/using_bootstrap/bootstrap_classes_requirement.png) from [Pathwrigth](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59540308/))
- [x] In the navbar you used plain text instead of using the input element, please use a real input, it doesn't have look just like the original but similar, see the inline comment bellow for more clarification
![image](https://user-images.githubusercontent.com/3244654/81469711-d4633b00-91e6-11ea-8f33-3954c5a9835c.png)
- [x] In the 'Subscribe' section on large screens there is a alignment issue, also the left content should be aligned in height with the right content; the image from right content is not displaying please fix that
#### Clone: 
![image](https://user-images.githubusercontent.com/3244654/81469792-50f61980-91e7-11ea-8361-b51732eaff6c.png)
#### Original:
![image](https://user-images.githubusercontent.com/3244654/81469838-90246a80-91e7-11ea-9f1e-04e9293255f6.png)
### <em>Kindly fix the REQUIRED issues and send another code review request.

Cheers and Happy coding! :clap: 

##### Code review by Alexandru Bangau

(Feel free to contact me on slack by the handle of alexandru if you have any additional question)</em>

### :stop_sign:  Status: Changes required.

rammazzoti2000 reviewed (changes requested):
#### :stop_sign: Status: Required Changes

8Bts committed:
Replaced span with input on navbar, Fixed subscribe section bugs, replaced possible css properties with bootsrtap classes

8Bts committed:
Readme update

8Bts committed:
Fixed stylelint errors

8Bts committed:
Fixed another stylelint errors

8Bts commented:
Thanks for review @rammazzoti2000 and helping me to make this project better. I fixed the problems you mentioned.

- [x] Enabled stickler

- [x] Changed `<span>` element to `<input>` element. Even defined transition duration and width change on  `:focus` state.

- [x] Fixed bugs on the subscribe section.

- [x] Added even more bootstrap classes instead of using CSS. Even removed declarations for grouped elements and added them on HTML one by one. 

### For the next code reviewers 
Please don't ask me to use more bootstrap classes again. This is the maximum usage for my design. Some bootstrap classes can't be responsive and I needed to define custom stylings inside medium and large size breakpoint media queries.
I know that my CSS file looks too big for the Bootstrap based project, but if you check the live version, you can see that I tried my best for making a page that looks very similar to the actual one. In order to achieve this, I needed more CSS code lines.

lekegitrepo commented:
#### Hi @8Bts :+1: 
#### You did a good job by implementing the changes required in the last review. This project is approved.

#### Note: As mentioned by the previous reviewer; try to make the height of the left and right contents in the subscribe section to be the same.

### Reviewed by Temitayo Adeleke.

- Github: [@lekegitrepo](https://github.com/lekegitrepo)
- Twitter: [@leketemi](https://twitter.com/leketemi)
- Linkedin: [linkedin](https://www.linkedin.com/in/temitayo-adeleke/)

lekegitrepo reviewed (approved):
#### Status: Approved.

8Bts committed:
Fixed height problem of subscribe section

8Bts committed:
Fixed stylelint issues
